kubernetes/nodecollection.go

```diff
@@ -86,6 +86,39 @@ func (cns ClientsetNodeSource) NodeAddress(node *v1.Node) (string, int32, error)
 	return "", 0, fmt.Errorf("Could not find internal IP address for node %s ", node.Name)
 }
 
+// Endpoint an enumeration representing the various metrics endpoints
+type Endpoint uint8
+
+const (
+	// NodeStatsSummaryEndpoint the /stats/summary endpoint
+	NodeStatsSummaryEndpoint Endpoint = iota
+
+	// NodeContainerEndpoint the /stats/containers endpoint
+	NodeContainerEndpoint
+
+	// NodeCadvisorEndpoint the /metrics/cadvisor/prometheus endpoint
+	NodeCadvisorEndpoint
+)
+
+// EndpointMask a map representing the currently active endpoints.
+// The keys of the map are the currently active endpoints.
+type EndpointMask map[Endpoint]struct{}
+
+// SetAvailable sets an endpoint availability state according to the supplied boolean
+func (m EndpointMask) SetAvailable(endpoint Endpoint, available bool) {
+	if available {
+		m[endpoint] = struct{}{}
+	} else {
+		delete(m, endpoint)
+	}
+}
+
+// GetAvailable gets the availability of an endpoint
+func (m EndpointMask) GetAvailable(endpoint Endpoint) bool {
```

Comment:
🤠 We don't put Get in our Getter names around these here parts pardner 🌟 🌅

Reply:
🤠 👍